src/main/scala/uk/gov/nationalarchives/tdr/api/graphql/DeferredResolver.scala

```diff
@@ -17,10 +17,15 @@ class DeferredResolver extends sangria.execution.deferred.DeferredResolver[Consi
         consignmentId.map(
           id => context.consignmentService.getConsignmentFileProgress(id)
         ).getOrElse(Future.successful(0))
+      case DeferParentFolder(consignmentId) =>
+        consignmentId.map(
+          id => context.consignmentService.getConsignmentParentFolder(id)
+        ).get
```

Comment:
Is the `.get` necessary, now that this case returns an optional value?

Reply:
I did initially try without the `.get` but kept getting a type mismatch error:
![Screenshot from 2020-10-22 09-38-48](https://user-images.githubusercontent.com/49832549/96847064-a139fe80-144a-11eb-8927-d80f65721b03.png)
But if I change the return type for the deferred resolver then the other case statements have type mismatches. There might be a better way to get around it though that I'm not thinking of!

Reply:
Oh, that's interesting. I didn't realise it was `Option[Future[String]]`. Probably easiest to have a quick call to decide how to handle this - give me a shout on Slack.

Reply:
Sure will do! :+1: